ui/app/controllers/vault/cluster/secrets/backends.js

```diff
@@ -2,36 +2,84 @@
  * Copyright (c) HashiCorp, Inc.
  * SPDX-License-Identifier: MPL-2.0
  */
-
-import { filterBy } from '@ember/object/computed';
-import { computed } from '@ember/object';
+/* eslint ember/no-computed-properties-in-native-classes: 'warn' */
 import Controller from '@ember/controller';
-import { task } from 'ember-concurrency';
-import { supportedSecretBackends } from 'vault/helpers/supported-secret-backends';
 import { inject as service } from '@ember/service';
-const LINKED_BACKENDS = supportedSecretBackends();
-
-export default Controller.extend({
-  flashMessages: service(),
-  displayableBackends: filterBy('model', 'shouldIncludeInList'),
-
-  supportedBackends: computed('displayableBackends', 'displayableBackends.[]', function () {
-    return (this.displayableBackends || [])
-      .filter((backend) => LINKED_BACKENDS.includes(backend.get('engineType')))
-      .sortBy('id');
-  }),
-
-  unsupportedBackends: computed(
-    'displayableBackends',
-    'displayableBackends.[]',
-    'supportedBackends',
-    'supportedBackends.[]',
-    function () {
-      return (this.displayableBackends || []).slice().removeObjects(this.supportedBackends).sortBy('id');
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { filterBy } from '@ember/object/computed';
+import { dropTask } from 'ember-concurrency';
+
+export default class VaultClusterSecretsBackendController extends Controller {
+  @service flashMessages;
+  @filterBy('model', 'shouldIncludeInList') displayableBackends;
+
+  @tracked secretEngineOptions = [];
+  @tracked selectedEngineType = null;
+  @tracked selectedEngineName = null;
+
+  get sortedDisplayableBackends() {
+    // show supported secret engines first and then organize those by id.
+    const sortedBackends = this.displayableBackends.sort(
+      (a, b) => b.isSupportedBackend - a.isSupportedBackend || a.id - b.id
+    );
+
+    // return an options list to filter by engine type, ex: 'kv'
+    if (this.selectedEngineType) {
+      // check first if the user has also filtered by name.
+      if (this.selectedEngineName) {
+        return this.filterByName(sortedBackends);
+      }
+      // otherwise filter by engine type
+      return this.filterByEngineType(sortedBackends);
+    }
+
+    // return an options list to filter by engine name, ex: 'secret'
+    if (this.selectedEngineName) {
+      return this.filterByName(sortedBackends);
     }
-  ),
+    // no filters, return full sorted list.
+    return sortedBackends;
+  }
+
+  filterByName(backendList) {
+    return backendList.filter((backend) => this.selectedEngineName === backend.get('id'));
+  }
+
+  filterByEngineType(backendList) {
```

Comment:
[non-blocking] Since we're not reusing `filterByName` and `filterByEngineType` (and their internal logic is one line) I almost think `sortedDisplayableBackends` would be easier to read if we returned the filtered list directly and skipped calling these helper fns